src/test/regress/expected/tableam.out

```diff
@@ -0,0 +1,299 @@
+SHOW server_version \gset
+SELECT substring(:'server_version', '\d+')::int > 11 AS server_version_above_eleven
+\gset
+\if :server_version_above_eleven
+\else
+\q
+\endif
+SET citus.shard_replication_factor to 1;
+SET citus.next_shard_id TO 60000;
+SET citus.next_placement_id TO 60000;
+SET citus.shard_count TO 4;
+create schema test_tableam;
+set search_path to test_tableam;
+SELECT public.run_command_on_coordinator_and_workers($Q$
+	CREATE FUNCTION fake_am_handler(internal)
+	RETURNS table_am_handler
+	AS 'citus'
+	LANGUAGE C;
+	CREATE ACCESS METHOD fake_am TYPE TABLE HANDLER fake_am_handler;
+$Q$);
+ run_command_on_coordinator_and_workers
+---------------------------------------------------------------------
+
+(1 row)
+
+--
+-- Hash distributed table using a non-default table access method
+--
+create table test_hash_dist(id int, val int) using fake_am;
+insert into test_hash_dist values (1, 1);
+WARNING:  fake_tuple_insert
+select create_distributed_table('test_hash_dist','id');
+WARNING:  fake_scan_getnextslot
+CONTEXT:  SQL statement "SELECT EXISTS (SELECT 1 FROM test_tableam.test_hash_dist)"
+WARNING:  fake_scan_getnextslot
+NOTICE:  Copying data from local table...
+WARNING:  fake_scan_getnextslot
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$test_tableam.test_hash_dist$$)
+ create_distributed_table
```

Comment:
optional: you could possibly do something like `ALTER EXTENSION seg ADD ACCESS METHOD fake_am;` to check whether the extension gets propagated.